### PR TITLE
fix: add request userID, drop mocks

### PR DESCRIPTION
**Pull request by sethyates**
## Description of this change

Add request user ID and drop mocks

## Why is this change being made?
- [ ] Chore (non-functional changes)
- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

N/A

## Checklist

- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [X] I have informed stakeholders of my changes.

### CODE REVIEW

File: request/context.go [hunk 1]
```diff
@@ -33,30 +33,29 @@ type Key string
...
 
-// HighCardinalityKeys is a map of high-cardinality keys
-var HighCardinalityKeys = map[Key]bool{
-	IDKey:         true,
-	OperationKey:  false,
-	TimestampKey:  true,
-	TenantKey:     false,
-	URLKey:        true,
-	OriginatorKey: false,
+// LowCardinalityKeys is a map of high-cardinality keys
```
Comment (algorithmcardboard) [inline request/context.go:54]: LowCardinalityKeys is a map of high-cardinality keys

File: request/context.go [hunk 2]
```diff
@@ -33,30 +33,29 @@ type Key string
...
+var LowCardinalityKeys = map[Key]bool{
+	OperationKey:  true,
```
Comment (algorithmcardboard) [inline request/context.go:56]: IIRC operation includes path parameters and hence may not be low cardinality

**Review by algorithmcardboard — APPROVED** [review]

File: request/context.go (hunk 1 above)
Comment (sethyates) [reply request/context.go:54]: How so?

Comment (algorithmcardboard) [reply request/context.go:54]: The comment line needs fixing?

File: request/context.go (hunk 2 above)
Comment (sethyates) [reply request/context.go:56]: IIRC, it is the `operationId` from openAPI and the service+method from grpc. But I'll check. `URLKey` is definitely the path parameters

File: request/context.go [hunk 3]
```diff
@@ -134,6 +135,21 @@ func RequireID(ctx context.Context) (string, error) {
...
+
+// User returns the User ID or an empty string
+func User(ctx context.Context) string {
+	return Value[string](ctx, UserKey)
+}
+
+// RequireUser returns the request User or an error if not set
+func RequireUser(ctx context.Context) (string, error) {
+	return RequireValue[string](ctx, UserKey, errors.Forbidden)
+}
```
Comment (suejungshin) [inline request/context.go:151]: cool!

**Pull request merged**